### PR TITLE
Revamped shared reactive infrastructure

### PR DESCRIPTION
**Pull request by vasilisp**
This PR enhances our reactive infrastructure. Notably, we provide:

- `Eliom_shared`, which is an updated version of `Eliom_csreact`. `Eliom_shared` provides *shared* versions of the `React` and `ReactiveData` modules. `Eliom_shared.React` and `Eliom_shared.ReactiveData` operate (via shared functions) on signals and data that have both a server-side and a client-side representation.

- TyXML-based server-side `Eliom_content.{Html5,Svg}.R` modules, which accept shared signals and data. See ocsigen/tyxml#76 for the TyXML side of things. The implementation is in the `Eliom_shared_content` module (which is not exported).

Once this gets merged, we will be close to a new Eliom release.

### CODE REVIEW

File: src/lib/eliom_shared.eliom [hunk 1]
```diff
@@ -0,0 +1,677 @@
...
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ *)
+
+{shared{
+
+   (* put this in Lwt_react? Find a better name? *)
```
Comment (Drup) [inline src/lib/eliom_shared.eliom]: Yes please. `from_lwt` sounds like a good name.

File: src/lib/eliom_shared_content.eliom [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+    let unoption_string () =
+      {shared#{ Html5_f.Wrapped_functions.unoption_string }}
+
+  end
+
+  module R = struct
+
+    (* Same as the SVG version, with Svg -> Html5 and `SVG ->
+       `HTML5. We can in principle functorize, but probably not worth
+       the trouble. Make sure they stay synced! *)
```
Comment (Drup) [inline src/lib/eliom_shared_content.eliom]: 100% chance someone is going to break that. Functorize it. It's not even really hard and it's shorter.

**Comment by Drup** [comment]
The amount of boilerplate in `Eliom_shared` is slightly unsettling. There are also redefinitions which I don't understand the purpose. I'll read more carefully later.

**Comment by vasilisp** [comment]
@Drup (and everyone else): you can still provide comments, and I will address them in master before the release.

**Pull request merged**